server/src/test/java/org/opensearch/indices/replication/OngoingSegmentReplicationsTests.java

```diff
@@ -228,4 +229,47 @@ public void testShardAlreadyReplicatingToNode() throws IOException {
         replications.prepareForReplication(request, segmentSegmentFileChunkWriter);
         assertThrows(OpenSearchException.class, () -> { replications.prepareForReplication(request, segmentSegmentFileChunkWriter); });
     }
+
+    public void testStartReplicationWithNoFilesToFetch() throws IOException {
+        // create a replications object and request a checkpoint.
+        OngoingSegmentReplications replications = spy(new OngoingSegmentReplications(mockIndicesService, recoverySettings));
+        final CheckpointInfoRequest request = new CheckpointInfoRequest(
+            1L,
+            replica.routingEntry().allocationId().getId(),
+            replicaDiscoveryNode,
+            testCheckpoint
+        );
+        // mock the FileChunkWriter so we can assert its ever called.
+        final FileChunkWriter segmentSegmentFileChunkWriter = mock(FileChunkWriter.class);
+        // Prepare for replication step - and ensure copyState is added to cache.
+        final CopyState copyState = replications.prepareForReplication(request, segmentSegmentFileChunkWriter);
+        assertTrue(replications.isInCopyStateMap(request.getCheckpoint()));
+        assertEquals(1, replications.size());
+        assertEquals(1, copyState.refCount());
+
+        getSegmentFilesRequest = new GetSegmentFilesRequest(
+            1L,
+            replica.routingEntry().allocationId().getId(),
+            replicaDiscoveryNode,
+            Collections.emptyList(),
+            testCheckpoint
+        );
+
+        // invoke startSegmentCopy and assert our fileChunkWriter is never invoked.
+        replications.startSegmentCopy(getSegmentFilesRequest, new ActionListener<>() {
+            @Override
+            public void onResponse(GetSegmentFilesResponse getSegmentFilesResponse) {
+                assertEquals(Collections.emptyList(), getSegmentFilesResponse.files);
+                assertEquals(0, copyState.refCount());
+                assertFalse(replications.isInCopyStateMap(request.getCheckpoint()));
+                verifyNoInteractions(segmentSegmentFileChunkWriter);
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                logger.error("Unexpected failure", e);
+                Assert.fail();
```

Comment:
Nitpick - it would be convenient to include a message in the assertion failure:

```
Assert.fail("Unexpected failure: " + e);
```

Reply:
updated.